packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-menus.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState, useCallback } from 'react';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import semver from 'semver';
+import {
+  Button,
+  Icon,
+  Menu,
+  MenuItem,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+import type { RootState } from '../../../modules';
+import { newPipelineFromText } from '../../../modules/import-pipeline';
+import { openCreateView } from '../../../modules';
+import { saveCurrentPipeline } from '../../../modules/saved-pipeline';
+import { savingPipelineOpen } from '../../../modules/saving-pipeline';
+import { setIsNewPipelineConfirm } from '../../../modules/is-new-pipeline-confirm';
+import { VIEWS_MIN_SERVER_VERSION } from '../../../constants';
+
+const buttonStyles = css({
+  marginLeft: spacing[1],
+  marginRight: spacing[1],
+});
+type PipelineActionMenuProp<ActionType> = {
+  onAction: (action: ActionType) => void;
+  title: string;
+  glyph: string;
+  menuItems: { title: string; action: ActionType }[];
+};
+function PipelineActionMenu<T>({
+  onAction,
+  title,
+  glyph,
+  menuItems,
+}: PipelineActionMenuProp<T>) {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const onMenuItemClick = useCallback(
+    (evt) => {
+      evt.stopPropagation();
+      setIsMenuOpen(false);
+      onAction(evt.currentTarget.dataset.action);
+    },
+    [onAction]
+  );
+
+  return (
+    <Menu
+      open={isMenuOpen}
+      setOpen={setIsMenuOpen}
+      justify="start"
+      trigger={({
+        onClick,
+        children,
+      }: {
+        onClick(): void;
+        children: React.ReactChildren;
+      }) => (
+        <Button
+          title={title}
+          aria-label={title}
+          className={buttonStyles}
+          variant="primary"
+          size="xsmall"
+          leftGlyph={<Icon glyph={glyph} />}
+          onClick={(evt) => {
+            evt.stopPropagation();
+            onClick();
+          }}
+        >
+          {title}
+          {isMenuOpen && children}
```

Comment:
IIRC you don't need to do this yourself, it should be ok to always render children and not remove them from the component tree manually (this is probably why the closing animation is broken right now)